### PR TITLE
Updated the most recent pinned trilinos version

### PR DESCRIPTION
**Pull request by ddement**

### CODE REVIEW

**Comment by PaulMullowney** [comment]
Does Nalu build/run successfully on crusher and frontier with this version of Trilions?

**Comment by ddement** [comment]
@PaulMullowney This (along with a pull request we have up for Nalu-Wind) allows us to build and run with this version of Trilinos, as well as trilinos@develop on Frontier.  The patches we have to apply to STK will be incorporated into our Trilinos snapshot soon.

File: repos/exawind/packages/trilinos/package.py [hunk 1]
```diff
@@ -30,6 +30,8 @@ class Trilinos(bTrilinos, SMCMakeExtension):
     variant("pic", default=True,
             description="Position independent code")
 
+    depends_on('cmake@3.23.1:', type='build')
```
Comment (psakievich) [inline repos/exawind/packages/trilinos/package.py]: I think this should be given a `when` condition for the latest trilinos.

**Comment by ddement** [comment]
@psakievich We've tried a few permutations, but it keeps failing CI's concretization test for cee due to the cmake change.  It has been working locally from the start on ascicgpu057, so there's something subtle that I may be missing.  Perhaps something on CI itself needs tweaking?  Any ideas on how to get it passing?

**Comment by psakievich** [comment]
I'm not sure yet. I'll try to take a look. Keep pinging me if I don't get to it in the next few days.

File: configs/cee/packages.yaml [hunk 2]
```diff
@@ -9,8 +9,6 @@ packages:
     version: [1.80.0]
   cmake:
     externals:
-    - spec: cmake@3.22.2
-      prefix: /projects/wind/system-spack/opt/spack/linux-rhel7-x86_64/gcc-9.3.0/cmake-3.22.2-s67tgoaqaw4ky5kmpuzxdg5une7xouvh/
     - spec: cmake@3.25.0%clang@13.0.1~doc+ncurses+ownlibs~qt build_type=Release arch=linux-rhel7-x86_64
```
Comment (psakievich) [inline configs/cee/packages.yaml]: I think this issue is the arch in these external specs. The ubuntu runner is not rhel7-x86-64 so it is likely failing b/c the specs are too restrictive
```suggestion
    - spec: cmake@3.25.0%clang@13.0.1~doc+ncurses+ownlibs~qt build_type=Release
```

File: configs/cee/packages.yaml [hunk 3]
```diff
@@ -9,8 +9,6 @@ packages:
...
       prefix: /projects/wind/core-spack-manager/scripts/admin-system-setup/packages/.spack-env/view/clang/13.0.1/cmake/3.25.0/flff6w2l
     - spec: cmake@3.25.0%gcc@10.3.0~doc+ncurses+ownlibs~qt build_type=Release arch=linux-rhel7-x86_64
```
Comment (psakievich) [inline configs/cee/packages.yaml]:
```suggestion
    - spec: cmake@3.25.0%gcc@10.3.0~doc+ncurses+ownlibs~qt build_type=Release
```

File: configs/spock/packages.yaml [hunk 4]
```diff
@@ -69,14 +69,14 @@ packages:
 #        prefix: /opt/cray/pe/parallel-netcdf/1.12.1.7/amd/4.3
 #        modules:
 #          - cray-parallel-netcdf/1.12.1.7
-  cmake:
```
Comment (psakievich) [inline configs/spock/packages.yaml:72]: @jrood-nrel are you okay with this?

Comment (ddement) [reply configs/spock/packages.yaml:72]: Isn't Spock decommissioned? (check the warning at the top of the page here: https://docs.olcf.ornl.gov/systems/spock_quick_start_guide.html).  Perhaps we should just remove support for the platform.

Comment (jrood-nrel) [reply configs/spock/packages.yaml:72]: Removing Spock is fine with me.

**Comment by psakievich** [comment]
@ddement we're going to merge this soon. Just deciding on how we want to handle keeping the challenge problem work pinned to the current trilinos release we've been using. @alanw0 we are proposing pinning our work to the old release we're having success with, and tasking the stk team with working through issues that pop up on summit/frontier from updating trilinos.

**Comment by ddement** [comment]
@psakievich Sounds good to me.  We are in the process of snapshotting a couple of new STK improvements into Trilinos (including removing the need for a couple of the patches).  If those go through in a timely manner, it might be good to change the latest pinned version to one that includes them.  Otherwise, we can just update it in a future PR once everything is in place.

**Pull request closed without merging**